### DISCUSSION TITLE
SoundTransform always present (?)

### DISCUSSION THREAD

**Discussion by 2jfw** [airsdk/Adobe-Runtime-Support#1925] [Q&A] [answered]
@ajwfrost I was wondering why every `Sprite` (and thus nearly all extending `DisplayObjects`) would have an existing `soundTransform`-Object?

Is this actually required?
I would somehow think that this is overhead and should only be instantiated when the programmer/user really needs it.

![grafik](https://user-images.githubusercontent.com/73781224/169519985-acc77138-0e37-422d-b74b-90129ec92a70.png)

**Comment by ajwfrost** [comment, marked as answer]
Hi

This is a property that's actually only instantiated when you query for it.... including the queries that a debugger would make, per your above screenshot. A new `flash.media.SoundTransform` object is created and is populated with the transform details of any sound that's associated with the sprite - or left with default values, which is what you're seeing in the above.

So this isn't overhead in normal operation, it's just created on-demand. Interestingly, so is the `graphics` property of a `Sprite`.... so whenever you access this, a new `Graphics` object is created and stored on the sprite object to then be returned in future.

cheers

**Reply by 2jfw** [reply]
Thanks for the explanation and insight, Andrew! Much appreciated
